packages/compiler/src/experimental/typekit/index.ts

```diff
@@ -1,2 +1,187 @@
+import { compilerAssert, Program } from "../../core/index.js";
+import { Realm, REALM_TYPEKIT } from "../realm.js";
+import { Typekit, TypekitPrototype } from "./define-kit.js";
+
 export * from "./define-kit.js";
 export * from "./kits/index.js";
+
+/**
+ * Create a new Typekit that operates in the given realm.
+ *
+ * Ordinarily, you should use the default typekit `$` to manipulate types in the current program, or call `$` with a
+ * Realm or Program as the first argument if you want to work in a specific realm or in the default typekit realm of
+ * a specific program.
+ *
+ * @param realm - The realm to create the typekit in.
+ *
+ * @experimental
+ */
+export function createTypekit(realm: Realm): Typekit {
+  const tk = Object.create(TypekitPrototype);
+
+  const handler: ProxyHandler<Typekit> = {
+    get(target, prop, receiver) {
+      if (prop === "program") {
+        // don't wrap program (probably need to ensure this isn't a nested program somewhere)
+        return realm.program;
+      }
+
+      if (prop === "realm") {
+        return realm;
+      }
+
+      const value = Reflect.get(target, prop, receiver);
+
+      if (typeof value === "function") {
+        return function (this: any, ...args: any[]) {
+          return value.apply(proxy, args);
+        };
+      }
+
+      if (typeof value === "object" && value !== null) {
+        return new Proxy(value, handler);
+      }
+
+      return value;
+    },
+  };
+
+  const proxy = new Proxy(tk, handler);
```

Comment:
The proxy lives, phew.

Reply:
It's still necessary to carry the correct, root `this` binding of the typekit forward through the access chain. I tried some things with property descriptors and it didnt work, all roads led back to Proxy.